### PR TITLE
Update release version to 0.5.0

### PR DESCRIPTION
**Pull request by aarondav**

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,6 +1,52 @@
...
+    .. code:: python
+
+        import mlflow
+        run = mlflow.start_run()
+        mlflow.tracking.get_service().set_terminated(run.info.run_uuid)
+
+    (Note that in a future version, we will likely rename or alias ``run_uuid``.)
+
+  - The Python entities returned by the servive API are now accessible in ``mlflow.entities`` directly. Where previously you may have used ``mlflow.entities.experiment.Experiment``, you would now just use ``mlflow.entities.Experiemnt``. The previous version still exists, but is deprecated and may be hidden in a future version.
+- url-parse dependency updated to 1.4.3 due to CVE in previous versions (#292, @mateiz)
```
Comment (mateiz) [inline CHANGELOG.rst]: This is not a breaking change; it's in JS

File: CHANGELOG.rst [hunk 2]
```diff
@@ -1,6 +1,52 @@
...
+
+    .. code:: python
+
+        import mlflow
+        run = mlflow.start_run()
+        mlflow.tracking.get_service().set_terminated(run.info.run_uuid)
+
+    (Note that in a future version, we will likely rename or alias ``run_uuid``.)
+
+  - The Python entities returned by the servive API are now accessible in ``mlflow.entities`` directly. Where previously you may have used ``mlflow.entities.experiment.Experiment``, you would now just use ``mlflow.entities.Experiemnt``. The previous version still exists, but is deprecated and may be hidden in a future version.
```
Comment (smurching) [inline CHANGELOG.rst]: servive -> REST?

File: CHANGELOG.rst [hunk 3]
```diff
@@ -1,6 +1,51 @@
...
+0.5.0 (2018-08-17)
+------------------
+
+MLflow 0.5.0 includes some major improvements, including Keras and PyTorch first-class support as models, SFTP support as an artifactory, an improved scatterplot visualization to compare runs, and a more complete Python SDK for low-level experiment and run management.
+
+Breaking changes:
+
+- The Tracking API has been split into two pieces, a "basic logging" and a "tracking service" API. The "basic logging" API deals with logging metrics, parameters, and artifacts to the currently-active active run, and is accessible in ``mlflow`` (e.g., ``mlflow.log_param``). The tracking service API allow managing experiments and runs (especially historical runs) and is available in ``mlflow.tracking``. The tracking service API will look analogous to the upcoming R and Java Tracking Service SDKs. Please be aware of the following breaking changes:
+
+  - ``mlflow.tracking`` no longer exposes the fluent API, only ``mlflow``. So, code that was written like ``from mlflow.tracking import log_param`` will have to be ``from mlflow import log_param`` (note that almost all examples were already doing this).
```
Comment (mateiz) [inline CHANGELOG.rst]: Do we still want to use the word fluent here?

File: CHANGELOG.rst [hunk 4]
```diff
@@ -1,6 +1,51 @@
...
 =========
 
+0.5.0 (2018-08-17)
+------------------
+
+MLflow 0.5.0 includes some major improvements, including Keras and PyTorch first-class support as models, SFTP support as an artifactory, an improved scatterplot visualization to compare runs, and a more complete Python SDK for low-level experiment and run management.
+
+Breaking changes:
+
+- The Tracking API has been split into two pieces, a "basic logging" and a "tracking service" API. The "basic logging" API deals with logging metrics, parameters, and artifacts to the currently-active active run, and is accessible in ``mlflow`` (e.g., ``mlflow.log_param``). The tracking service API allow managing experiments and runs (especially historical runs) and is available in ``mlflow.tracking``. The tracking service API will look analogous to the upcoming R and Java Tracking Service SDKs. Please be aware of the following breaking changes:
```
Comment (mateiz) [inline CHANGELOG.rst]: `a "basic logging" and a "tracking service" API` -> `a "basic logging" API and a "tracking service" API`

File: CHANGELOG.rst [hunk 5]
```diff
@@ -1,6 +1,51 @@
...
+
+  - ``mlflow.tracking`` no longer exposes the fluent API, only ``mlflow``. So, code that was written like ``from mlflow.tracking import log_param`` will have to be ``from mlflow import log_param`` (note that almost all examples were already doing this).
+  - Access to the service API goes through the ``mlflow.tracking.get_service()`` function, which relies on the same tracking server set by either the environment variable ``MLFLOW_TRACKING_URI`` or by code with ``mlflow.tracking.set_tracking_uri()``. So code that used to look like ``mlflow.tracking.get_run()`` will now have to do ``mlflow.tracking.get_service().get_run()``. This does not apply to the fluent API.
+  - ``mlflow.ActiveRun`` has been converted into a lightweight wrapper around ``mlflow.entities.Run`` to enable the Python ``with`` syntax. This means that there are no longer any special methods on the object` returned when calling ``mlflow.start_run()``. These can be converted to the service API. For example, ``mlflow.start_run().set_terminated()`` would now be called as:
```
Comment (mateiz) [inline CHANGELOG.rst]: Extra \` after `object`

File: CHANGELOG.rst [hunk 6]
```diff
@@ -1,6 +1,51 @@
...
+
+    .. code:: python
+
+        import mlflow
+        run = mlflow.start_run()
+        mlflow.tracking.get_service().set_terminated(run.info.run_uuid)
+
+    (Note that in a future version, we will likely rename or alias ``run_uuid``.)
```
Comment (mateiz) [inline CHANGELOG.rst]: Not sure that we need the example or the note here. I don't think people were calling methods on the returned objects super often.

File: CHANGELOG.rst [hunk 7]
```diff
@@ -1,6 +1,51 @@
...
+- [Artifacts] SFTP artifactory store added (#260, @ToonKBC)
+- [Sagemaker] Users can specify a custom VPC when deploying SageMaker models (#304, @dbczumar)
+- Pyfunc serialization now includes the Python version, and warns if the major version differs (can be suppressed by using ``load_pyfunc(suppress_warnings=True)``) (#230, @dbczumar)
+- Pyfunc serve/predict will activate conda environment stored in MLModel. This can be disabled by adding ``--no-conda`` to ``mlflow pyfunc serve`` or ``mlflow pyfunc predict`` (#225, @0wu)
+- Python SDK formalized in ``mlflow.tracking``. This includes adding SDK methods for ``get_run``, ``list_experiments``, ``get_experiment``, and ``set_terminated``. (#299, @aarondav)
+- ``mlflow run`` can now be run against projects with no ``conda.yaml`` specified. By default, an empty conda environment will be created -- previously, it would just fail. You can still pass ``--no-conda`` to avoid entering a conda environment altogether (#218, @smurching)
+
+Bug fixes:
+
+- Fix numpy array array serialization for int64 and other related types, allowing pyfunc to return such results (#240, @arinto)
```
Comment (mateiz) [inline CHANGELOG.rst]: array array

File: CHANGELOG.rst [hunk 8]
```diff
@@ -1,6 +1,43 @@
...
+  - Access to the service API goes through the ``mlflow.tracking.get_service()`` function, which relies on the same tracking server set by either the environment variable ``MLFLOW_TRACKING_URI`` or by code with ``mlflow.tracking.set_tracking_uri()``. So code that used to look like ``mlflow.tracking.get_run()`` will now have to do ``mlflow.tracking.get_service().get_run()``. This does not apply to the basic logging API.
+  - ``mlflow.ActiveRun`` has been converted into a lightweight wrapper around ``mlflow.entities.Run`` to enable the Python ``with`` syntax. This means that there are no longer any special methods on the object returned when calling ``mlflow.start_run()``. These can be converted to the service API.
+
+  - The Python entities returned by the servive API are now accessible in ``mlflow.entities`` directly. Where previously you may have used ``mlflow.entities.experiment.Experiment``, you would now just use ``mlflow.entities.Experiemnt``. The previous version still exists, but is deprecated and may be hidden in a future version.
```
Comment (mparkhe) [inline CHANGELOG.rst]: Experiment typo

File: docs/source/tracking.rst [hunk 9]
```diff
@@ -156,6 +158,21 @@ environment variable.
         mlflow.log_param("a", 1)
```
Comment (mparkhe) [inline docs/source/tracking.rst]: documentation for `log_param` in `fluent` says that value should be a `string` (or maybe we should document that param value gets cast to string in service)

**Comment by aarondav** [comment]
@mparkhe thanks, comments addressed!

While building the docs, I noticed that our test-requirements for PyTorch seemed to be pulling a Linux-only wheel, so I changed it to pip -- let's see if tests pass.

I also fixed up a bunch of references in the docs that I noticed were messed up.

**Pull request merged**